### PR TITLE
[design] use Inter font on Android

### LINKED ISSUE #6633: Bold instead of italic font on iOS for text wrapped in tildes
***Type***: Bug

***Summary***: Text wrapped in tildes displayed with the bold font on iOS instead of italic. 
Note: text wrapped in asterisks displays with the bold font as well - issue is that we need an italic for text in tildes.

**Note**: it's all fine on Android.
On desktop app there is no font at all applied but text is parsed (i.e. the `~text~` and `*bold*` displayed without asterisks/tildes). Related issue for desktop https://github.com/status-im/status-react/issues/5275

#### Expected behavior
Italic font applied to for text wrapped in tildes on iOS

#### Actual behavior
Bold font applied to for text wrapped in tildes on iOS

screenshot 2018-11-07 at 15 28 26

### Reproduction

- Open Status and create new account
- Navigate to any chat (e.g. create unique public chat) 
- Send `~I'm italic~` and verify there is italic font applied to the text

### Additional Information
* Status version: 0.9.30 (11730) 
* Operating System: iOS (12.0.1)

### LINKED ISSUE #7738: Update to latest Inter UI Typeface
# Problem
Current release uses an old version of Inter UI. We want this to be updated to the latest version.

# Implementation
New version needs to be downloaded and installed across platforms. https://github.com/rsms/inter

# Acceptance Criteria

# Notes

# Future Steps

### PR DESCRIPTION
**Pull request by yenda**
Fix #7738 
Fix #6633

- implement typography component
- replace InterUI font by Inter font
- fix italic in iOS

# Review
@errorists can you check the android screenshots?

most important changes:
- `components/src/status_im/ui/components/react.cljs` text and text-input are modified
- `src/status_im/ui/components/typography.cljs` new typography component with sane defaults

# Next step
typography defaults and colors can be defined in a json file that is imported at startup time

# Testing
Changes all the fonts so testing this revolves around checking for blatant design flaws 
 (removed screenshots for visibility)

status: ready

### CODE REVIEW

File: src/status_im/ui/components/common/styles.cljs [hunk 1]
```diff
@@ -137,10 +134,9 @@
    :align-items      :center
    :justify-content  :center})
 
-(defnstyle counter-label [size]
+(defn counter-label [size]
   {:font-size  (/ size 2)
    :color      colors/white
-   :android    {:line-height (+ (/ size 2) 2)}
```
Comment (errorists) [inline src/status_im/ui/components/common/styles.cljs:143]: won't removing that break things? it looked like it did something important xD

File: src/status_im/ui/components/list/styles.cljs [hunk 2]
```diff
@@ -150,7 +138,7 @@
    :margin-left  6
    :margin-right 4
    :font-size    11
-   :font-weight  :bold})
+   :font-weight  700})
```
Comment (errorists) [inline src/status_im/ui/components/list/styles.cljs]: so, wondering here, does it work without curly quotes? I thought RN demanded that font-weight should be specified like '500', '600', '700' etc?

File: src/status_im/ui/components/typography.cljs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        (assoc :font-family (str font-family "-"
+                                 (case font-weight
+                                   400 "Regular"
+                                   500 "Medium"
+                                   600 "SemiBold"
+                                   700 "Bold")
+                                 (when (= font-style :italic)
+                                   "Italic"))
+               :line-height line-height)
+        (dissoc :font-weight :font-style))))
```
Comment (errorists) [inline src/status_im/ui/components/typography.cljs]: ❤️ this is awesome!! great work Eric

**Comment by errorists** [comment]
One thing I've noticed, [on this view](https://user-images.githubusercontent.com/1181225/54401577-96dd4d80-46c8-11e9-8e6a-40df37ce59a4.png) it looks like Welcome to Status doesn't agree to be a `Header`, it should :)

**Comment by errorists** [comment]
image
on iOS I've found this weirdness in wallet balance

**Comment by hesterbruikman** [comment]
> 
> 
> image
> on iOS I've found this weirdness in wallet balance

@errorists I might be wrong, but I think this has to do with an error bar to cover for the case of currency information not being received for whatever reason. @goranjovic can probably confirm or say what else might be going on. for context:
https://github.com/status-im/status-react/issues/7641
https://github.com/status-im/status-react/issues/5356

File: src/status_im/ui/components/common/styles.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/ui/components/common/styles.cljs:143]: the line height is automatically added to text fields based on the font-size now

File: src/status_im/ui/components/list/styles.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/ui/components/list/styles.cljs]: because in typography a font-family is selected based on font-weight and font-style

**Comment by yenda** [comment]
@hesterbruikman not related
@errorists fixed

Screenshot_1552642701
Screenshot_1552642774

**Comment by goranjovic** [comment]
@hesterbruikman looks like a bug 
~@errorists  Can you please create a bug issue? what are the reproduction conditions? just ios and this font?~
or just fix here if introduced in this PR

File: src/status_im/chat/commands/impl/transactions.cljs [hunk 4]
```diff
@@ -39,7 +39,7 @@
     [react/view transactions-styles/asset-main
      [react/image {:source (-> asset :icon :source)
                    :style  transactions-styles/asset-icon}]
-     [react/text {:style transactions-styles/asset-symbol}
+     [react/text {}
```
Comment (flexsurfer) [inline src/status_im/chat/commands/impl/transactions.cljs]: why do we need an empty map?

File: src/status_im/chat/commands/impl/transactions/styles.cljs [hunk 5]
```diff
@@ -71,8 +71,9 @@
 
 (defn command-send-amount-text
   [outgoing]
-  {:font-size 22
-   :color     (if outgoing colors/white colors/blue)})
+  {:font-size   22
+   :font-weight 600
```
Comment (flexsurfer) [inline src/status_im/chat/commands/impl/transactions/styles.cljs]: is it possible to use both number and strings?

File: src/status_im/ui/screens/chat/message/message.cljs [hunk 6]
```diff
@@ -99,14 +93,16 @@
...
+       (cond-> {:style (style/text-message collapsible? outgoing)}
+         (and collapsible? (not expanded?))
+         (assoc :number-of-lines constants/lines-collapse-threshold))
        (if-let [render-recipe (:render-recipe content)]
          (chat.utils/render-chunks render-recipe message)
          (:text content))
-       [react/text {:style (style/message-timestamp-placeholder-text outgoing)}
-        (timestamp-with-padding timestamp-str)]]
+       [react/text {:style {:color (if outgoing colors/blue colors/blue-light)}}
+        (str " AAAAA")]]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/chat/message/message.cljs]: could you elaborate ` (str " AAAAA")` ?

**Review by flexsurfer — COMMENTED** [review]
i'm not sure about empty map for react/text

File: src/status_im/chat/commands/impl/transactions/styles.cljs (hunk 5 above)
Comment (yenda) [reply src/status_im/chat/commands/impl/transactions/styles.cljs]: the react-native property "fontWeight" needs to be a string on react-native, however this is used to build a "fontFamily" in the `typography` namespace and is dissoc from the style. I replaced all the fontWeight integer by strings in a new commit that is coming soon

File: src/status_im/ui/screens/chat/message/message.cljs (hunk 6 above)
Comment (yenda) [reply src/status_im/ui/screens/chat/message/message.cljs]: It's an invisible field to place the timestamp properly, spaces are trimmed so they wouldn't work

Comment (flexsurfer) [reply src/status_im/ui/screens/chat/message/message.cljs]: hm, how is it invisible? it even has a colour

Comment (errorists) [reply src/status_im/ui/screens/chat/message/message.cljs]: @yenda if we are going to such lengths to have the timestamp in place, maybe, a different design for the timestamp is required altogether? they only really position themselves correctly like 50% of the time, often it's just hanging weird.

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/chat/message/message.cljs (hunk 6 above)
Comment (yenda) [reply src/status_im/ui/screens/chat/message/message.cljs]: @errorists yeah that would be great I was thinking about asking you that yesterday. The main issue being that it seems impossible to get text wrapping and alignment of timestamp on the right of the last line of text without overlapping

**Review by vitvly — APPROVED** [review]

**Comment by churik** [comment]
@yenda 
#### 1) Tests are failed because text on buttons 'CREATE NEW ACCOUNT', 'I ALREADY HAVE AN ACCOUNT', 'ADD EXISTING ACCOUNT' is not capitalized anymore.
What is the reason to change this?

#### 2) Error when you tap on Profile > Devices


**OS:** IOS, Android

#### 3) Error when I tap on Cryptostrikers under Collectibles in wallet.
You should have some Cryptostrikers to see this error.
Can't reproduce with others collectibles, i.e. with EthermonAsset


**OS:** IOS, Android

#### 4) `Amount` in /send command is not aligned with other elements
фьщ

**OS:** IOS

#### 5) Line for links is dashed
link124

**OS:** IOS

#### 6) Text formatting with ~~ and ** make text black for sender and  ~~ formatting doesn't make text italic
dsfsdfsdf

**OS:** IOS, Android

**Comment by yenda** [comment]
@churik I can't reproduce any of those can you try with latest build once it is ready?

**Comment by yenda** [comment]
@churik @errorists and yeah all the buttons and stuff are lowercase now so we have to update tests. It was confirmed by @errorists

**Comment by errorists** [comment]
👋 I'll just chime in, the goal is to unify the designs of both Android and iOS as much as possible, visually. There is no reason to have All Caps buttons on Android other than maintain appearance of platform consistency so I'd much rather have them lowercase. With regards to platform specific patterns, it's either all in or none at all, we shouldn't pick and choose where it looks like Android and where not :) Hope it helped.

**Comment by antdanchenko** [comment]
@yenda, @churik please wait for me to update all texts in tests before merging the PR, I am going to run nightly build against this PR, will take longer then used to

**Comment by churik** [comment]
@yenda 4 and 5 on IOS 11.4 are still relevant to me. Can you please take a look?
All other issues are fixed.

**Comment by yenda** [comment]
@churik which model of iOS are you using? I don't have this on 6s plus

**Comment by churik** [comment]
@yenda 
with last commit ( https://github.com/status-im/status-react/commit/cbf879f513004123f3d5f09582d6463ab4e129d8 ) I think that 4 can be considered as fixed.
Now it looks:
аааа

5 is still there: 
centered

Same situation in Enter URL Field in Status browser.
UrK

Device: IPhone 7
OS: 11.4.1

Please ping me when PR should be retested. Thanks!

**Comment by yenda** [comment]
@churik please check last build, looks like 5 is fixed on iphone 6 plus now

**Comment by churik** [comment]
@annadanchenko @Serhy appreciate if it can be briefly checked on the small device and on IPhone X.
Mostly checks for screens + input fields are needed.
I checked on Android 8.0 (LG V20) and IOS 11.4 (IPhone 7)

**Comment by yenda** [comment]
@antdanchenko how are you doing with the e2e tests?

**Comment by antdanchenko** [comment]
@yenda I am on it, should be ready today

**Comment by antdanchenko** [comment]
@yenda please rebase the PR on develop

**Comment by annadanchenko** [comment]
@yenda on build 18 for iOS (real iphone 5s):
### issue 7 Invalid arity error if send request to a contact from wallet or from chat


Steps:
for account with a contact
1. Open Wallet -> Receive -> Send a Transaction request
2. Select recipient from Contacts list and provide valid amount, e.g. 1
3. Tap on Send Request
4. or open chat with a contact and send a request of 1 STT to it

### issue 8 Picture is left aligned on welcome screen

**Comment by annadanchenko** [comment]
@yenda 
### issue 9 Invalid arity error if open Transaction history with one sent transaction (ETH, on ropsten)



Steps:
- switch to ropsten with upstream RPC network
- send a transaction of ETH from wallet to anyone (can be a contact or scanned QR code)
- open Transaction history

**Comment by yenda** [comment]
@annadanchenko regarding 8 it's already in develop I think, because it can't be related to this PR

**Comment by yenda** [comment]
@annadanchenko the 2 others should be fixed

**Comment by yenda** [comment]
@annadanchenko I would say this introduced the 8 https://github.com/status-im/status-react/pull/7755

**Comment by Serhy** [comment]
With the latest builds on iOS (v12.1.4) and Android (v8.1) each time I navigate to 1-1 chat with the sent `/request` or `/send` messages (so sending the `/send` / `/request` too) - app crashes with error:

Screenshot 2019-03-19 at 16 21 58


- Open Status and create new account
- Add contact to start 1-1 chat
- Send `/request ETH 0` message in 1-1 chat

**Comment by Serhy** [comment]
Please ignore above https://github.com/status-im/status-react/pull/7742#issuecomment-474393084
I see it reported already

**Comment by yenda** [comment]
@Serhy should be fixed now
@annadanchenko I also fixed the italic on iOS

**Comment by Serhy** [comment]
### Issue 10

Once device (Android / iOS) switches to Mobile network - app crashes with:
Screenshot 2019-03-19 at 19 13 43


```

03-19 19:05:03.046  2981  2981 D ConnectivityService: filterNetworkStateForUid() uid: 1000 networkInfo: [type: MOBILE[LTE] - MOBILE, state: CONNECTED/CONNECTED, reason: connected, extra: internet, failover: false, available: true, roaming: false]
03-19 19:05:03.046  2981  4887 D ConnectivityService: filterNetworkStateForUid() uid: 11321 networkInfo: [type: MOBILE[LTE] - MOBILE, state: CONNECTED/CONNECTED, reason: connected, extra: internet, failover: false, available: true, roaming: false]
03-19 19:05:03.046  2981  3176 D ConnectivityService: filterNetworkStateForUid() uid: 10022 networkInfo: [type: MOBILE[LTE] - MOBILE, state: CONNECTED/CONNECTED, reason: connected, extra: internet, failover: false, available: true, roaming: false]
03-19 19:05:03.047  3616  3957 D AECLog  : AECModule: onReceive: android.net.conn.INET_CONDITION_ACTION
03-19 19:05:03.047  3616  3957 D AECLog  : AECModule: handleMessage: HANDLE_ENTITLEMENT_CONNECTIVITY_CHANGE
03-19 19:05:03.047  4587  4675 I zygote  : Deoptimizing void com.facebook.react.uimanager.UIImplementation.manageChildren(int, com.facebook.react.bridge.ReadableArray, com.facebook.react.bridge.ReadableArray, com.facebook.react.bridge.ReadableArray, com.facebook.react.bridge.ReadableArray, com.facebook.react.bridge.ReadableArray) due to JIT inline cache
03-19 19:05:03.047  3376  3527 D NetworkController.MobileSignalController(0/1): updateConnectivity: mCurrentState.bluetoothTethered + false
03-19 19:05:03.047  2981  3176 I chatty  : uid=1000(system) ConnectivityThr identical 2 lines
03-19 19:05:03.047  4587  4674 E ReactNativeJS: Error: Invalid arity: 3
03-19 19:05:03.047  4587  4674 E ReactNativeJS: 
03-19 19:05:03.047  4587  4674 E ReactNativeJS: This error is located at:
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent98
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent398
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in settings-sheet
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent392
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in n
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent391
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent138
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent102
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent101
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in bottom-sheet
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent87
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent2
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in root
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in reagent1
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in RCTView
03-19 19:05:03.047  4587  4674 E ReactNativeJS:     in t
03-19 19:05:03.047  3376  3376 E SignalClusterView: setSlotFocusVisible : subId 1 visible false
03-19 19:05:03.047  3376  3376 E SignalClusterView: setSlotFocusVisible : subId 1 visible false

```

**Comment by yenda** [comment]
@Serhy 10 should be fixed now

**Comment by annadanchenko** [comment]
> @Serhy 10 should be fixed now

@yenda I still can reproduce issue 10 - crash if mobile data is used and app is launched in build 25 (ios, iphone 5s) https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7742/25/

**Comment by annadanchenko** [comment]
@yenda issue 11, iOS only: if edit url field after DApp/site was opened then url is higher than mid of the field. Can reproduce on iphone 7plus, iphone 5s

Steps:
1. Open any Dapp
2. edit url

**Comment by yenda** [comment]
@antdanchenko it seems like some e2e are failing due to things that are already in develop, how come they got merged without passing the tests?

**Comment by antdanchenko** [comment]
@yenda  `test_backup_recovery_phrase_warning_from_wallet` is failing due to #7750, it just stopped working, the issues is not related to any PR

`test_send_two_transactions_one_after_another_in_dapp` and ` test_send_eth_in_1_1_chat` should be fixed by me in this PR, they are not failing in any other

`test_search_chat_on_home` has been developed before separation dapps and chat screen, but merged after. It was already fixed in #7772

**Comment by yenda** [comment]
@antdanchenko thanks for the details

@annadanchenko the error regarding mobile network was another one, fixed as well now. Alignment is fixed too

**Comment by antdanchenko** [comment]
@yenda, please, ignore this comment

**Comment by antdanchenko** [comment]
both tests are expected to fail, `test_search_chat_on_home` is already fixed in develop and `test_backup_recovery_phrase_warning_from_wallet` should be fixed in #7750

**Comment by Serhy** [comment]
### Issue 12

Desktop (MacOS) app crashes when opening Profile screen
[StatusMacOSCrashLog.log](https://github.com/status-im/status-react/files/2988369/statusMacOSCrash.log)
Screenshot 2019-03-20 at 14 47 19

- Open Status
- Log in to the existing account
- Click `Profile` tab

**Comment by annadanchenko** [comment]
Ignore, this issue exists on develop too

on Windows build 33 https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7742/33/

White screen is shown after account is created. If close app and launch again then can sign in and see main screen fine.
[Status (1).log](https://github.com/status-im/status-react/files/2988613/Status.1.log)
screen

**Comment by annadanchenko** [comment]
@yenda please ignore windows issue with blank screen it's on develop too https://github.com/status-im/status-react/issues/7776

**Comment by annadanchenko** [comment]
@Serhy @yenda  I've tried Windows build 33 https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7742/33/ and see that font is different on PR vs nightly. Taking into account that desktop will get mobile UI soon I think it's fine. No more issues found on Windows
in PR:
text

in nightly:
nightly

**Comment by yenda** [comment]
@annadanchenko does it mean we can merge?

**Comment by Serhy** [comment]
@yenda give me 0.5h more please, final checks, no issues so far.

**Comment by yenda** [comment]
@Serhy Your time is up :D

**Comment by Serhy** [comment]
Don't see any new issues with iOS 12.1.4, Android 8.1, Android 6.0.1, MacOS. Let's merge @yenda

**Pull request merged**